modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/AbstractCreateIndexCommandBuilder.java

```diff
@@ -32,4 +33,10 @@ interface AbstractCreateIndexCommandBuilder<T extends AbstractIndexCommandBuilde
 
     /** List of the columns to index. There must be at least one column. */
     T columns(List<String> columns);
+
+    /** Initial status of the index. */
+    T status(CatalogIndexStatus status);
+
+    /** Flag indicating that this index has been created at the same time as its table. */
+    T isCreatedWithTable(boolean isCreatedWithTable);
```

Comment:
these two methods looks dangerous. I like the idea to create indexes in AVAILABLE status, but I believe this should be done internally. Otherwise someone will accidentally create AVAILABLE index on non empty table. Therefore I propose to remove these methods from CommandBuilder.